### PR TITLE
fix: relax upper bound for urllib dependency

### LINKED ISSUE #4362: Remove upper bound on urllib in the requirements
**Describe the bug**
https://github.com/aws/sagemaker-python-sdk/issues/4168 previously fixed this.

The changes to the `setup.py` in https://github.com/aws/sagemaker-python-sdk/commit/68ff700cf923244400852dd8265c37f580748640#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R67-R72 reintroduced the old versions of urllib. Relax the restriction on the version. 

I've also opened https://github.com/aws/sagemaker-python-sdk/issues/4361 for the pinned versions of fast api and uvcorn separately.

**To reproduce**
This was introduced in release 2.199.0. 

`python -m pip install sagemaker==2.203.0` results in unresolvable deps since urilib is restricted to above version two in many other projects.

**Expected behavior**
Install to succeed, project to be clear of CVE from transitive deps.

**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.203.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: n/a
- **Framework version**: n/a
- **Python version**: 3.12/3.11/3.10/3.9
- **CPU or GPU**: cpu
- **Custom Docker image (Y/N)**: N

**Additional context**

### PR DESCRIPTION
**Pull request by jmahlik**
Closes #4362

*Issue #, if available:* #4362

*Description of changes:*
Relax the urllib deps to allow using newer versions. Boto supports them as well.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mohanasudhan — APPROVED** [review]

**Review by mohanasudhan — APPROVED** [review]

**Comment by jmahlik** [comment]
@mohanasudhan am I able to /bot run all once it is approved? Or is that locked to maintainers?

**Comment by mohanasudhan** [comment]
Its locked to maintainers. We are taking deeper in the code-build failures and planning to revamp. Apologies for inconvenience.

**Comment by jmahlik** [comment]
Totally understand why it is locked down. Doesn't seem like the bot worked there. Just let me know if a rebase is needed again happy to do it.

**Review by knikure — COMMENTED** [review]
/bot run all

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-notebook-tests
* Commit ID: 8da8b49bc8e0bd874be4355a2e1b4f29e43d2c62
* Result: SUCCEEDED
* [Build Logs](https://bkqkixfb59.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c8fbb820-cebf-4b4c-8dc1-f4fc5add1c6c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-local-mode-tests
* Commit ID: 8da8b49bc8e0bd874be4355a2e1b4f29e43d2c62
* Result: SUCCEEDED
* [Build Logs](https://8huhkqo601.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5fb41a2a-05ae-4645-b348-035bd3be169f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-unit-tests
* Commit ID: 8da8b49bc8e0bd874be4355a2e1b4f29e43d2c62
* Result: FAILED
* [Build Logs](https://fegcnxu6da.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=deaa64c3-649c-414f-a277-a8d7b4affe51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-pr
* Commit ID: 8da8b49bc8e0bd874be4355a2e1b4f29e43d2c62
* Result: FAILED
* [Build Logs](https://abkt2mlh3k.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=982e896c-cee0-4361-b7b4-3491ec3676f7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-slow-tests
* Commit ID: 8da8b49bc8e0bd874be4355a2e1b4f29e43d2c62
* Result: SUCCEEDED
* [Build Logs](https://omfrydptb9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b7622eb3-17b9-4251-9c24-73a105acf8d2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jmahlik** [comment]
It looks like these are unrelated flakey tests. Thoughts on merging this or should I try to rebase?

**Comment by jmahlik** [comment]
If it is easier for y'all to make the change, I can close this MR. The urllib restriction is causing some pretty big issues downstream.

**Review by benieric — COMMENTED** [review]
/bot run all

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-pr
* Commit ID: 259ed3ede6cff28299e20ea49f8cbca37e978ae4
* Result: SUCCEEDED
* [Build Logs](https://abkt2mlh3k.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=fe544e1e-6af8-4855-baa0-7dba821e6a32%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-notebook-tests
* Commit ID: 259ed3ede6cff28299e20ea49f8cbca37e978ae4
* Result: SUCCEEDED
* [Build Logs](https://bkqkixfb59.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cfd103a7-4554-4111-a886-3310d496ced4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-slow-tests
* Commit ID: 259ed3ede6cff28299e20ea49f8cbca37e978ae4
* Result: SUCCEEDED
* [Build Logs](https://omfrydptb9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=18ede3c8-5cc6-480e-ba4e-1ced6f4746aa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-unit-tests
* Commit ID: 259ed3ede6cff28299e20ea49f8cbca37e978ae4
* Result: SUCCEEDED
* [Build Logs](https://fegcnxu6da.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5c9add85-f32e-4d94-8d43-8894f6a307b8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by benieric — COMMENTED** [review]
/bot run local-mode-tests

**Comment by mufaddal-rohawala** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-local-mode-tests
* Commit ID: 259ed3ede6cff28299e20ea49f8cbca37e978ae4
* Result: SUCCEEDED
* [Build Logs](https://8huhkqo601.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=481f778f-0589-421c-ac38-2e4dd1c8fc51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by knikure — APPROVED** [review]

**Pull request merged**
